src/System.Device.Gpio/System/Device/Gpio/Drivers/LibGpiodDriver.Linux.cs

```diff
@@ -122,7 +220,45 @@ protected internal override void SetPinMode(int pinNumber, PinMode mode)
 
         protected internal override WaitForEventResult WaitForEvent(int pinNumber, PinEventTypes eventType, CancellationToken cancellationToken)
         {
-            throw new NotImplementedException();
+            if (!_pinNumberToEventHandler.TryGetValue(pinNumber, out LibGpiodDriverEventHandler eventHandler))
+            {
+                eventHandler = new LibGpiodDriverEventHandler(pinNumber, new CancellationTokenSource());
+                if (_pinNumberToEventHandler.TryAdd(pinNumber, eventHandler))
+                {
+                    eventHandler.PinHandle = SubscribeForEvent(pinNumber, eventType);
+                    eventHandler.Task = InitializeEventDetectionTask(eventHandler, eventHandler.CancellationTokenSource.Token);
+                }
+            }
+
+            CancellationTokenSource cancelWhenEventOccured = new CancellationTokenSource();
```

Comment:
could this be just a bool?

Reply:
yes it is, good catch

Reply:
oh I was mistaken, didn't work with Boolean, seems cannot use value type

Reply:
@buyaa-n in the function you will likely need to use ref bool